### PR TITLE
Remove unused variables from OMP build

### PR DESCRIPTION
**Pull request by pbartholomew08**
There are unused dummy arguments, but they may be required by generic interface, or are not yet implemented.

These variables were found by setting CMAKE_Compiler_FLAGS to
```
-g -std=f2008 -Wall -Wpedantic -Werror -Wimplicit-interface -Wimplicit-procedure -Wno-unused-dummy-argument
```
in the CMake configuration.

We should aim to do development with (similar) flags to ensure code quality.

### CODE REVIEW

**Comment by pbartholomew08** [comment]
I realise I've probably broken the "open an issue" workflow, but wanted to show why we should consider enabling these flags. If you think it's worth doing/discussing there then I'll close and do things properly.

**Comment by Nanoseb** [comment]
Yes, happy with that. We should consider making these flags the default at least in debug mode with GNU.

**Comment by slaizet** [comment]
agreed!

**Review by semi-h — APPROVED** [review]

**Comment by pbartholomew08** [comment]
I'll add these default options and see if we have equivalents for Nvidia in Xcompact3d.

**Comment by Nanoseb** [comment]
Sorry one last thing. It seems the github actions aren't testing the debug build (not sure which one is the default). So we might want to add `-DCMAKE_BUILD_TYPE=Debug` in the config line there.

**Comment by semi-h** [comment]
Can we still use below to activate debug mode?
https://github.com/xcompact3d/x3d2/blob/c6cc95b1b637cdea1990894371d15eff0ee4c0eb/docs/source/developer/build.rst?plain=1#L16

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -43,8 +43,11 @@ jobs:
         echo "$GITHUB_WORKSPACE/openmpi-${OMPI_VERSION}/installed/bin" \
           >> $GITHUB_PATH
 
+    - name: Check Debug build
+      run: FC=mpif90 cmake -S . -B build -DCMAKE_BUILD_TYPE=Debug
```
Comment (Nanoseb) [inline .github/workflows/build_and_test.yml:47]: I think this is just configuring cmake, but not running the compilation itself. If github actions are run in order of the yaml file the release configuration will be used for the build itself.

**Comment by pbartholomew08** [comment]
Note I had to change how some coefficients were stored in the OMP tridiagonal solver. I think this is because they were being set as a local variable. This would work within a `PARALLEL PRIVATE(du_s, du_e)` region, however with multiple `SIMD` regions I'm not sure how the local storage works. I've gone with replacing this with explicit storage of size `SZ` (the vector length?) which should cost no more than the assumed vector-private storage I think.

**Comment by semi-h** [comment]
> Note I had to change how some coefficients were stored in the OMP tridiagonal solver. I think this is because they were being set as a local variable. This would work within a `PARALLEL PRIVATE(du_s, du_e)` region, however with multiple `SIMD` regions I'm not sure how the local storage works. I've gone with replacing this with explicit storage of size `SZ` (the vector length?) which should cost no more than the assumed vector-private storage I think.

Well spotted! I copied this from the CUDA version, added simd loops and obviously forgot to fix this bit. I don't think `simd` region can itself extend the single variable on the left to a vector sized one and store it that way, the way it was coded is just wrong. Maybe this was even preventing vectorisation, haven't really checked.

**Comment by pbartholomew08** [comment]
Spotted by the compiler - not me!

**Comment by pbartholomew08** [comment]
I believe this is good to go. I've added additional debug flags to the NVHPC build based on 2decomp. For either backend we can select between these by setting CMAKE_BUILD_TYPE either at initial configuration or using `ccmake`

**Pull request merged**